cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -469,6 +469,28 @@ BITWISE_NOT(int64)
 #undef DATE_TYPES
 #undef NUMERIC_BOOL_DATE_TYPES
 #undef NUMERIC_DATE_TYPES
+
+#define SIGN(TYPE)                         \
+  FORCE_INLINE                             \
+  gdv_##TYPE sign_##TYPE(gdv_##TYPE in1) { \
+    gdv_##TYPE out;                        \
+    if (in1 > 0) {                         \
+      out = static_cast<gdv_##TYPE>(1);    \
+    } else if (in1 < 0) {                  \
+      out = static_cast<gdv_##TYPE>(-1);   \
+    } else {                               \
+      out = static_cast<gdv_##TYPE>(in1);  \
```

Comment:
cast is unnecessary here

Reply:
But if it is int we need to return 1 or -1 and if it is float or double I need to return 1.0 or -1.0

Reply:
i mean in the else block. in1 type is return type

Reply:
Ok got it.